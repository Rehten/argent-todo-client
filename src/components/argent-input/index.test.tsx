import { ArgentInput } from './index';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import React from "react";
import TestRenderer from 'react-test-renderer';

describe('ArgentInput', () => {
    let container: HTMLElement;

    beforeEach(() => {
        // подготавливаем DOM-элемент, куда будем рендерить
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        // подчищаем после завершения
        unmountComponentAtNode(container);
        container.remove();
        container = (null as any) as HTMLElement;
    });

    it('should be object', () => {
        expect(ArgentInput).toBeDefined();
        expect(ArgentInput).not.toBeNull();
        expect(ArgentInput).toBeInstanceOf(Object);
    });

    it('should containt HTML input node', () => {
        act(() => {
            render(<ArgentInput />, container);
        });

        const htmlInput: HTMLInputElement | null = container.querySelector('input');

        expect(htmlInput).not.toBeNull();
    });

    it('should have div wrapper with data-test id', () => {
        const testMarker: string = 'Qwerty';

        act(() => {
            render(<ArgentInput dataTest={testMarker} />, container);
        });

        const divWrapper: HTMLDivElement | null = container.querySelector(`div[data-test]`);

        expect(divWrapper).not.toBeNull();
        expect(divWrapper?.dataset.test).toEqual(testMarker);
    });

    it('should be same as snapshot', () => {
        const rslt = TestRenderer.create(<ArgentInput dataTest="Snapshot Testing" />);
        expect(rslt).toMatchSnapshot();
    });
});
