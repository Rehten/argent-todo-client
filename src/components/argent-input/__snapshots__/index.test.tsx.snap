// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgentInput should be same as snapshot 1`] = `
<div
  data-test="Snapshot Testing"
>
  <input
    className="form-control"
  />
</div>
`;
